bento_wes/db.py

```diff
@@ -198,97 +252,113 @@ def run_with_details_from_row(
                 run_id=run["id"],
                 state=run["state"],
                 request=run_request_from_row(run),
-                run_log=run_log_from_row(run, stream_content),
-                task_logs=cls.get_task_logs(c, run["id"]),
+                run_log=run_log_from_row(run, stream_content, self._settings),
+                task_logs=self.get_task_logs(self.c, run["id"]),
                 outputs=json.loads(run["outputs"]),
             )
         )
 
     @staticmethod
-    def _get_run_row(c: sqlite3.Cursor, run_id: uuid.UUID | str) -> sqlite3.Row | None:
+    def get_task_logs(c: sqlite3.Cursor, run_id: UUID | str) -> list:
```

Comment:
given our approach to cursors now where the objects never get reused, there's not much point in this being a passed argument.